12.21_다트게임.js

```diff
@@ -0,0 +1,34 @@
+function solution(dartResult) {
+    let arr1 = dartResult.split('');
+    let anwser = 0;
+    for (let i = 0; i < 12; i++) {
+        if (arr1[i] === '1' && arr1[i + 1] === '0') {
+            arr1.splice(i, 2, 10);
+        }
+    } //1,0 10으로 만들어주는거
+    for (let i = 0; i < 12; i++) {
+        if (typeof arr1[i] === 'string') {
+            switch (arr1[i]) {
+                case 'S':
+                    arr1.splice(i - 1, 2, arr1[i - 1] * 1);
+                    break;
+                case 'D':
+                    arr1.splice(i - 1, 2, arr1[i - 1] ** 2);
+                    break;
+                case 'T':
+                    arr1.splice(i - 1, 2, arr1[i - 1] ** 3);
+                    break;
+            }
+        } // 보너스 계산
```

Comment:
```suggestion
        switch (arr1[i]) {
            case 'S':
                arr1.splice(i - 1, 2, arr1[i - 1] * 1);
                break;
                case 'D':
                    arr1.splice(i - 1, 2, arr1[i - 1] ** 2);
                    break;
                case 'T':
                arr1.splice(i - 1, 2, arr1[i - 1] ** 3);
                break;
        }
```
이미 값이 S, D T 로 고정되어있기 때문에 ```switch~case```만으로도 풀릴 것 같습니다.